File: api/Models/Proposal.cs
using System;
using System.Collections.Generic;
using Azure;
using Azure.Data.Tables;

namespace api.Models;

public class Proposal : ITableEntity
{
    public Proposal(string rowKey, string partitionKey)
    {
        RowKey = rowKey;
        PartitionKey = partitionKey;
    }
    public string CustomerFullName { get; set; } = string.Empty;
    public string CustomerEmail { get; set; } = string.Empty;
    public string CustomerTelephoneNumber { get; set; } = string.Empty;
    public int ServiceType { get; set; }
    public int WarrantyType { get; set; }
    public int WarrantyQtd { get; set; }
    public int ExcecutionTime { get; set; }
    public string Power { get; set; } = string.Empty;
    public string ZipCode { get; set; } = string.Empty;
    public string Street { get; set; } = string.Empty;
    public string Neighborhood { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public List<Product> Products { get; set; } = new List<Product>();
    public int TotalPriceProducts { get; set; }
    public int LabourValue { get; set; }
    public int TotalPrice { get; set; }
    public string Notes { get; set; } = string.Empty;
    public string PartitionKey { get; set; } = string.Empty;
    public string RowKey { get; set; } = string.Empty;
    public string[] PaymentMethods { get; set; } = new string[] { };
    DateTimeOffset? ITableEntity.Timestamp { get; set; }
    ETag ITableEntity.ETag { get; set; }
}
public class Product
{
    public Product()
    {

    }
    public string Name { get; set; } = string.Empty;
    public int Quantity { get; set; }
}

File: app/src/app/interfaces/proposal.ts
import { Address, Customer } from "./customer";
import { Product } from "./product";
import { Service } from "./service";

export interface ProposalResponse {
  count: number;
  next: string;
  previous: string;
  proposals: ViewProposal[];
}
export interface ViewProposal {
  id: number;
  customerFullName: string;
  customerEmail: string;
  cellphone: string;
  power: string;
  price: number;
  date: string;
}
export interface Proposal {
  customerFullName: string;
  customerEmail: string;
  customerTelephoneNumber: string;
  serviceType: number;
  warrantyType: number;
  warrantyQtd: number;
  excecutionTime: number;
  power: string;
  zipCode: string;
  street: string;
  neighborhood: string;
  city: string;
  state: string;
  products: Product[];
  createdAt: Date;
  totalPriceProducts: number;
  labourValue: number;
  totalPrice: number;
  notes: string;
  partitionKey: string;
  rowKey: string;
  paymentMethod: string;
}


